lib/collection.js

```diff
@@ -45,7 +45,69 @@ module.exports = function Collection(db, state) {
     get writeConcern() { NotImplemented() },
 
     aggregate: NotImplemented,
-    bulkWrite: NotImplemented,
+    bulkWrite: function (operations, options, callback) {
+      const promises = []
+  
+      for (const operation of operations) {
+        let promise = null
+    
+        // Determine which operation to forward to
+        if (operation.insertOne) {
+          const { document } = operation.insertOne
+          promise = this.insertOne(document, options)
+        } else if (operation.updateOne) {
+          const { filter, update, ...opts } = operation.updateOne
+          promise = this.updateOne(filter, update, { ...options, ...opts })
+        } else if (operation.updateMany) {
+          const { filter, update, ...opts } = operation.updateMany
+          promise = this.updateMany(filter, update, { ...options, ...opts })
+        } else if (operation.deleteOne) {
+          const { filter } = operation.deleteOne
+          promise = this.deleteOne(filter, options)
+        } else if (operation.deleteMany) {
+          const { filter } = operation.deleteMany
+          promise = this.deleteMany(filter, options)
+        } else if (operation.replaceOne) {
+          const { filter, replacement, ...opts } = operation.replaceOne
+          promise = this.replaceOne(filter, replacement, { ...options, ...opts })
+        }
```

Comment:
What do you think about throwing on not supported operations instead of silently dropping them?

```suggestion
        } else {
          NotImplemented()
        }
```

Reply:
I think it's a good idea (that's what the driver does), but not the NotImplemented error, as it could confuse devs into thinking they wrote a good operation, but it's just not handled by mongo-mock yet. I went ahead and pulled the error message from the driver.

Any operations that are actually not implemented will already throw NotImplemented. I think the only one that was at risk at the time of me opening the PR was replaceOne.

relevant source: https://github.com/mongodb/node-mongodb-native/blob/main/src/bulk/common.ts#L1194